src/porepy/applications/complete_setups/terzaghi_biot.py

```diff
@@ -493,28 +520,36 @@ class PseudoOneDimensionalColumn(pp.ModelGeometry):
     params: dict
     """Simulation model parameters."""
 
+    def height(self) -> pp.number:
+        """Retrieve height of the domain, in scaled [m]."""
+        ls = 1 / self.units.m  # length scaling
+        height = self.params.get("height", 1.0)  # [m]
+        return height * ls
+
     def set_md_grid(self) -> None:
         """Create the mixed-dimensional grid based on two-dimensional Cartesian grid."""
-        height = self.params.get("height", 1.0)  # [m]
+        height = self.height()  # scaled [m]
+        phys_dims = np.array([height, height])  # scaled [m]
+
         num_cells = self.params.get("num_cells", 20)
-        ls = 1 / self.units.m
-        phys_dims = np.array([height, height]) * ls  # scaled [m]
         n_cells = np.array([1, num_cells])
+
+        sd: pp.Grid = pp.CartGrid(n_cells, phys_dims)
+        sd.compute_geometry()
+
         self.domain = pp.Domain(
             {
-                "xmin": 0,
-                "xmax": phys_dims[0],
-                "ymin": 0,
-                "ymax": phys_dims[1],
+                "xmin": 0,  # scaled [m]
```

Comment:
As a matter of taste, I think tagging "everything" with units is a bit too much.

Reply:
Sure. Fixed.